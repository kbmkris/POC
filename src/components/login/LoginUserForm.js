import React from "react";
import { PropTypes } from "prop-types";
import styles from "../../css/index.css";

/*
  For input elements, mention id as userData property name so one
  handleOnChange can be used
*/

const LoginUserForm = ({
  emailId,
  password,
  isLoading,
  handleOnChange,
  handleOnClickLogin,
  handleOnClickRegister
  }) =>  (
    <div className="main-div">
      <div className="login-div">
        <h3> Login Form </h3>
        <form className="login-form">
          <label
            className="label"
            htmlFor="emailId" >
            User Name :
          </label>
          <br />
          <input
            type="email"
            className="input"
            id="emailId"
            autoComplete="on"
            placeholder="Enter your email Id"
            value={emailId}
            onChange={handleOnChange} />
          <br />
          <label
            className="label"
            htmlFor="password" >
            Password :
          </label>
          <br />
          <input
            type="password"
            className="input"
            id="password"
            autoComplete="current-password"
            value={password}
            onChange={handleOnChange} />
          <br />
          <div className="button-div">
            <button
              type="submit"
              name="loginButton"
              className="button button-left"
              onClick={handleOnClickLogin} >
            {isLoading ? "Logging In..." : "Login"}
            </button>
            <button
              type="submit"
              name="registerButton"
              className="button button-right"
              onClick={handleOnClickRegister} >
              Register
            </button>
          </div>
        </form>
      </div>
    </div>
);


LoginUserForm.propTypes = {
  emailId : PropTypes.string.isRequired,
  password : PropTypes.string.isRequired,
  isLoading : PropTypes.bool.isRequired,
  handleOnChange : PropTypes.func.isRequired,
  handleOnClickLogin : PropTypes.func.isRequired,
  handleOnClickRegister : PropTypes.func.isRequired
};

LoginUserForm.defaultProps = {
  emailId : "",
  password : "",
  isLoading : false
};

export default LoginUserForm;
