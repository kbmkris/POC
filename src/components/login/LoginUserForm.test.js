import React from "react";
import LoginUserForm from "./LoginUserForm";
import renderer from "react-test-renderer";
import { mount, shallow } from "enzyme";

describe("LoginUserForm Test", () => {
  it("Snapshot - component should render", () => {
    const handleOnChange = jest.fn();
    const handleOnClickLogin = jest.fn();
    const handleOnClickRegister = jest.fn();
    const component = renderer
      .create(
        <LoginUserForm
          emailId = {""}
          password = {""}
          isLoading = {false}
          handleOnChange ={handleOnChange}
          handleOnClickLogin={handleOnClickLogin}
          handleOnClickRegister={handleOnClickRegister} />,
      );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe ("Dom validation using enzyme", () => {
  function setup (emailId, password, isLoading) {
    const props = {
      emailId : emailId,
      password : password,
      isLoading : isLoading,
      handleOnChange : jest.fn,
      handleOnClickLogin : jest.fn(),
      handleOnClickRegister : jest.fn()
    };
    return shallow(<LoginUserForm {...props} />);
  }
  it("check for all elements inside LoginUserForm", () => {
    const wrapper = setup("","",false);
    expect(wrapper.find("form").length).toBe(1);
    expect(wrapper.find("h3").text()).toEqual(" Login Form ");
    expect(wrapper.find("input").length).toBe(2);
    expect(wrapper.find("button").length).toBe(2);
    expect(wrapper.find("#emailId").length).toBe(1);
    expect(wrapper.find("#password").length).toBe(1);
    expect(wrapper.find("button").at(0).text()).toEqual("Login");
    expect(wrapper.find("button").at(1).text()).toEqual("Register");
  });
  it("check for login button value changes when loading", () => {
    const wrapper = setup("","",true);
    expect(wrapper.find("button").at(0).text()).toEqual("Logging In...");
   });
});
