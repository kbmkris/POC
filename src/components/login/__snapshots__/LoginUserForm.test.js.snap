// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginUserForm Test Snapshot - component should render 1`] = `
<div
  className="main-div"
>
  <div
    className="login-div"
  >
    <h3>
       Login Form 
    </h3>
    <form
      className="login-form"
    >
      <label
        className="label"
        htmlFor="emailId"
      >
        User Name :
      </label>
      <br />
      <input
        autoComplete="on"
        className="input"
        id="emailId"
        onChange={[MockFunction]}
        placeholder="Enter your email Id"
        type="email"
        value=""
      />
      <br />
      <label
        className="label"
        htmlFor="password"
      >
        Password :
      </label>
      <br />
      <input
        autoComplete="current-password"
        className="input"
        id="password"
        onChange={[MockFunction]}
        type="password"
        value=""
      />
      <br />
      <div
        className="button-div"
      >
        <button
          className="button button-left"
          name="loginButton"
          onClick={[MockFunction]}
          type="submit"
        >
          Login
        </button>
        <button
          className="button button-right"
          name="registerButton"
          onClick={[MockFunction]}
          type="submit"
        >
          Register
        </button>
      </div>
    </form>
  </div>
</div>
`;
